src/macro_util.rs

```diff
@@ -276,6 +407,32 @@ mod tests {
         assert_t_align_gteq_u_align!(u8, AU64, false);
     }
 
+    // remove this `cfg` when `size_of_val_raw` is stabilized
+    #[cfg(__INTERNAL_USE_ONLY_NIGHLTY_FEATURES_IN_TESTS)]
+    #[test]
+    fn test_trailing_field_offset() {
+        macro_rules! test {
+            (#[$cfg:meta] ($($ts:ty),* ; $trailing_field_ty:ty) => $expect:expr) => {{
+                #[$cfg]
+                struct Test($($ts,)* $trailing_field_ty);
+                assert_eq!(test!(@offset $($ts),* ; $trailing_field_ty), $expect);
+            }};
+            (#[$cfg:meta] $(#[$cfgs:meta])* ($($ts:ty),* ; $trailing_field_ty:ty) => $expect:expr) => {
+                test!(#[$cfg] ($($ts),* ; $trailing_field_ty) => $expect);
+                test!($(#[$cfgs])* ($($ts),* ; $trailing_field_ty) => $expect);
+            };
+            (@offset ; $_trailing:ty) => { trailing_field_offset!(Test, 0) };
+            (@offset $_t:ty ; $_trailing:ty) => { trailing_field_offset!(Test, 1) };
+        }
+
+        test!(#[repr(C)] #[repr(transparent)] #[repr(packed)] #[repr(C)] (; u8) => Some(0));
+        test!(#[repr(C)] #[repr(packed)] #[repr(C)] (u8; u8) => Some(1));
+        test!(#[repr(C)] (u8; [AU64]) => Some(8));
+        test!(#[repr(C)] (u8; Nested<u8, AU64>) => Some(8));
```

Comment:
Some other test ideas:
- `(u8; Nested<u8, [AU64]>)`
- `(; u8)`
- `(; [u8])`
- `(; AU64)`
- `(; [AU64])`
- `(; Nested<u8; [AU64]>)`

I'd also test with variants of `#[repr(packed(N))]` and `#[repr(align(N))]`, maybe for values of `N` in `{1, 2, 8, 16}`. I'm specifically interested in making sure that if the padding is modified by `repr(packed)`, the macro detects it. `repr(align)` shouldn't modify padding, and we should confirm that too.